File: LeetCodeProblemsGo/LeetCode1550ThreeConsecutiveOdds/LeetCode1550ThreeConsecutiveOdds.go
package main

func threeConsecutiveOdds(arr []int) bool {
	consecutiveOdds := 0
	for _, num := range arr {
		if num%2 == 1 {
			consecutiveOdds++
			if consecutiveOdds == 3 {
				return true
			}
			continue
		}
		consecutiveOdds = 0
	}
	return false
}

File: LeetCodeProblemsCMake/Problems/LeetCode1550ThreeConsecutiveOdds.cpp
//
// Created by saeed on 5/11/25.
//

#include "LeetCode1550ThreeConsecutiveOdds.h"

bool LeetCode1550ThreeConsecutiveOdds::threeConsecutiveOdds(std::vector<int> &arr) {
    int consecutiveOdds{};
    for (auto num : arr) {
        if (num % 2 == 0) {
            consecutiveOdds ++;
            if (consecutiveOdds ==3) {
                return true;
            }
            continue;
        }
        consecutiveOdds = 0;
    }
    return false;
}


